crates/fuel-gas-price-algorithm/src/v1.rs

```diff
@@ -307,11 +300,11 @@ impl core::ops::Deref for ClampedPercentage {
 impl AlgorithmUpdaterV1 {
     pub fn update_da_record_data(
         &mut self,
-        height_range: Range<u32>,
-        range_cost: u128,
+        heights: &[u32],
```

Comment:
It seems like all we do is iterating over the heights, and we do provide a range still in `simulation.rs`. Perhaps we should accept an `impl IntoIterator<Item = u32>` here to allow both ranges and slices to be provided.

No need to do this if you think it's overkill.

Reply:
I think it's overkill. What we'll get from the committer will be a `Vec`. The ranges are just fossils from the old code.

Reply:
Fair enough :+1: